time_manager/time_manager.F90

```diff
@@ -1213,9 +1213,27 @@ function real_to_time_type(x,err_msg) result(t)
   real :: tps
   real :: a
   tps = real(ticks_per_second)
-  a = x/spd
+
+  select type (x)
+  type is (real(kind=r4_kind))
+    a = x/spd
+  type is (real(kind=r8_kind))
+    a = x/spd
+  class default
+    call error_mesg('time_manager_mod::real_to_time_type', 'unsupported kind', FATAL)
```

Comment:
This error message should be more descriptive.  *"x is not one of the supported types of real(kind=4) or real(kind=8)."*